File: backend/spring-backend/src/main/java/com/florist/springbackend/dto/Purchase.java
package com.florist.springbackend.dto;

import com.florist.springbackend.Entity.Address;
import com.florist.springbackend.Entity.Customer;
import com.florist.springbackend.Entity.Order;
import com.florist.springbackend.Entity.OrderItem;
import lombok.Data;

import java.util.Set;

@Data
public class Purchase {
    private Customer customer;
    private Address shippingAddress;
    private Order order;
    private Set<OrderItem> orderItems;

}

File: frontend/angular-ecommerce/src/app/common/purchase/purchase.ts
import {Customer} from "../customer/customer";
import {Order} from "../order/order";
import {OrderItem} from "../order-item/order-item";
import {Address} from "../address/address";

export class Purchase {
  customer!: Customer;
  shippingAddress!: Address;
  order!: Order;
  orderItems!: OrderItem[];
}
